tools/picard.py

```diff
@@ -536,4 +536,35 @@ def execute(
             'RUN_BARCODE=' + run_barcode, 'LIBRARY_PARAMS=' + library_params
         ]
         PicardTools.execute(self, self.subtoolName, opts, JVMmemory)
+
+    def execute_single_sample(self, 
+        basecalls_dir,
+        output_file,
+        run_barcode, # this is a run-specific ID, ex. the flowcell ID
+        lane,
+        sample_alias,
+        picardOptions=None,
+        JVMmemory=None
+    ):
+        picardOptions = picardOptions or {}
+
+        opts_dict = self.defaults.copy()
+        for k, v in picardOptions.items():
+            opts_dict[k] = v
+        opts = []
+        for k, v in opts_dict.items():
+            if v is not None:
+                if type(v) in (list, tuple):
+                    for x in v:
+                        opts.append('='.join((k.upper(), str(x))))
+                else:
+                    opts.append('='.join((k.upper(), str(v))))
+        opts += [
```

Comment:
Thanks for setting `SAMPLE_ALIAS`. Can you also set `LIBRARY_NAME` to the same value? Our downstream iSNV code would appreciate it.

Reply:
Will do.

Reply:
`SAMPLE_ALIAS` and `LIBRARY_NAME ` are supposedly deprecated, but from what [the picard docs](https://broadinstitute.github.io/picard/command-line-overview.html#IlluminaBasecallsToSam) describe it sounds like picard expects barcode columns in the `LIBRARY_PARAMS` file (though I haven't yet confirmed that).

Reply:
Though maybe setting `BARCODE=N` would work?

Reply:
You've already set `SAMPLE_ALIAS` here, and it seems like Picard hasn't turned off these options yet.. given that it seems semantically wrong to send in a `LIBRARY_PARAMS` file in this scenario, hopefully they won't drop support for it. You could do `BARCODE=N` but you'd have to check whether Picard gets confused when there's no B in the `read_structure`.